nft_ingester/src/program_transformers/bubblegum/collection_verification.rs

```diff
@@ -1,88 +1,75 @@
-        upsert_asset_with_owner_and_delegate_info, upsert_asset_with_seq, upsert_collection_info,
-        upsert_collection_verified,
-    },
-};
+use crate::program_transformers::bubblegum::{upsert_asset_with_seq, upsert_collection_info};
 use blockbuster::{
     instruction::InstructionBundle,
     programs::bubblegum::{BubblegumInstruction, LeafSchema, Payload},
 };
+use log::debug;
+use mpl_bubblegum::state::metaplex_adapter::Collection;
 use sea_orm::query::*;
 
+use super::{save_changelog_event, upsert_asset_with_leaf_info};
+use crate::error::IngesterError;
 pub async fn process<'c, T>(
     parsing_result: &BubblegumInstruction,
     bundle: &InstructionBundle<'c>,
     txn: &'c T,
-    verify: bool,
 ) -> Result<(), IngesterError>
 where
     T: ConnectionTrait + TransactionTrait,
 {
-    if let (Some(le), Some(cl)) = (&parsing_result.leaf_update, &parsing_result.tree_update) {
-        // Do we need to update the `slot_updated` field as well as part of the table
-        // updates below?
+    if let (Some(le), Some(cl), Some(payload)) = (
+        &parsing_result.leaf_update,
+        &parsing_result.tree_update,
+        &parsing_result.payload,
+    ) {
+        let (collection, verify) = match payload {
+            Payload::CollectionVerification {
+                collection, verify, ..
+            } => (collection.clone(), verify.clone()),
+            _ => {
+                return Err(IngesterError::ParsingError(
+                    "Ix not parsed correctly".to_string(),
+                ));
+            }
+        };
+        debug!(
+            "Handling collection verification event for {} (verify: {}): {}",
+            collection, verify, bundle.txn_id
+        );
         let seq = save_changelog_event(cl, bundle.slot, txn).await?;
-        #[allow(unreachable_patterns)]
-        match le.schema {
-            LeafSchema::V1 {
-                id,
-                owner,
-                delegate,
-                ..
-            } => {
-                let id_bytes = id.to_bytes();
-                let owner_bytes = owner.to_bytes().to_vec();
-                let delegate = if owner == delegate {
-                    None
-                } else {
-                    Some(delegate.to_bytes().to_vec())
-                };
-
-                // Partial update of asset table with just leaf.
-                upsert_asset_with_leaf_info(
-                    txn,
-                    id_bytes.to_vec(),
-                    Some(le.leaf_hash.to_vec()),
-                    Some(seq as i64),
-                    false,
-                )
-                .await?;
-
-                // Partial update of asset table with just leaf owner and delegate.
-                upsert_asset_with_owner_and_delegate_info(
-                    txn,
-                    id_bytes.to_vec(),
-                    owner_bytes,
-                    delegate,
-                    seq as i64,
-                )
-                .await?;
```

Comment:
we could but it's unnecessary and i'd like to reduce the DB calls.

Reply:
Hmm, I am not sure because if we have:
- (seen and processed) update N mint in db
- (unseen) update N-1 owner change *not* in DB 
- (seen and processing) update N+2 collection verification in db

wouldnt it be better to process owner and delegate at N+2 when we have the data instead of watiing for N-1 to be backfilled at a later stage when it is seen ?

Reply:
In theory the total number of DB writes should be the same since N-1 when it is ingested would simply be skipped? So the overall number of DB calls remain similar or same?

Reply:
I could add it back in in a follow on PR if we want to have it before backfilled.  That can be standalone PR though